src/components/scenes/SpendingLimitsScene.tsx

```diff
@@ -42,10 +42,11 @@ export const SpendingLimitsScene = (props: Props) => {
     const isAuthorized = await account.checkPassword(password)
     if (!isAuthorized) return showToast(lstrings.password_check_incorrect_password_title)
 
+    const isZeroAmount = zeroString(transactionAmount)
     const spendingLimits = {
       transaction: {
-        isEnabled: transactionIsEnabled,
-        amount: parseFloat(transactionAmount)
+        isEnabled: transactionIsEnabled && !isZeroAmount,
+        amount: isZeroAmount ? 0 : parseFloat(transactionAmount)
```

Comment:
We don't want to turn off spending limits if the amount is 0. If the transaction limits are enabled, and the amount is 0, then we simply want to block every spend, regardless of amount.